### PR TITLE
mentorship_3: basic php

### PR DESCRIPTION
**Pull request by yaremkoP**

### CODE REVIEW

File: src/TextAnalyzer/Form.php [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            $this->setOutput($result);
+        }
+
+        $this->render($this->form);
+    }
+
+    /**
+     * @param string $form
+     */
+    public function setForm(string $form): void
```
Comment (vpashkov-cartera) [inline src/TextAnalyzer/Form.php:63]: YAGNI ?

File: src/TextAnalyzer/Form.php [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+        if ($form) {
+            $this->form = $form;
+        }
+    }
+
+    /**
+     * @param array $options
+     */
+    public function setOptions(array $options): void
```
Comment (vpashkov-cartera) [inline src/TextAnalyzer/Form.php:75]: same YAGNI

File: src/TextAnalyzer/Form.php [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+     */
+    public function render(string $form = '', ?array $options = null): void
+    {
+        if (isset($options)) {
+            $this->validateOptions($options);
+        }
+
+        $options = $options ?? $this->options;
+
+        if (strlen($form) > 0 && $this->validateForm($form)) {
```
Comment (vpashkov-cartera) [inline src/TextAnalyzer/Form.php]: having $form and $this->form is confusing, especially with two returns and return type of function  as a :void

File: src/TextAnalyzer/Form.php [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    /**
+     * @param string $form
+     * @return string
+     * @throws InvalidArgumentException
+     */
+    protected function validateForm(string $form = ''): string
+    {
+        if (strlen($form) === 0) {
```
Comment (vpashkov-cartera) [inline src/TextAnalyzer/Form.php:169]: I understand that you wanted to predict all possible cases and make the code bugless, but in my opinion that`s kind of "overengeneering".

File: src/TextAnalyzer/Form.php (hunk 3 above)
Comment (yaremkoPavlo) [reply src/TextAnalyzer/Form.php]: I applied more simpled solution

**Pull request merged**